substrate/frame/support/src/migrations.rs

```diff
@@ -880,6 +897,13 @@ impl<T: SteppedMigration> SteppedMigrations for T {
 		)
 	}
 
+	fn nth_migrating_prefixes(n: usize) 
+        -> Option<Result<impl IntoIterator<Item = Vec<u8>>, SteppedMigrationError>> {
+        n.is_zero()
+            .then(|| Ok(T::migrating_prefixes().into_iter().collect::<Vec<_>>()))
```

Comment:
Hm why is it allocating a vector here?  

This function should just be the identity function for `n == 0`.